Sources/Profile-Actions.php

```diff
@@ -967,4 +967,587 @@ function subscriptions($memID)
+		else
+		{
+			$data = $smcFunc['json_encode'](array(
+				'format' => $format,
+				'uid' => $memID,
+				'lang' => $context['member']['language'],
+				'included' => $included,
+				'start' => $start,
+				'latest' => $latest,
+				'datatype' => isset($current_datatype) ? $current_datatype : 'profile',
+			));
+
+			$smcFunc['db_insert']('insert', '{db_prefix}background_tasks',
+				array('task_file' => 'string-255', 'task_class' => 'string-255', 'task_data' => 'string', 'claimed_time' => 'int'),
+				array('$sourcedir/tasks/ExportProfileData.php', 'ExportProfileData_Background', $data, 0),
+				array()
+			);
+
+			// So the user can see that we've started.
+			if (!file_exists($realfilepath) && !file_exists($tempfilepath) && !file_exists($progressfile))
+			{
+				require_once($sourcedir . '/News.php');
+
+				$feed_meta = array_fill_keys(array('title', 'desc', 'source', 'self'), '');
+
+				buildXmlFeed($format, array(), $feed_meta, 'profile');
+				file_put_contents($tempfilepath, implode('', array($context['feed']['header'], $context['feed']['footer'])));
+
+				file_put_contents($progressfile, $smcFunc['json_encode'](array_fill_keys(array_keys($included), 0)));
+			}
+		}
+	}
+
+	createToken($context['token_check'], 'post');
+
+	$context['page_title'] = $txt['export_profile_data'];
+
+	if (empty($modSettings['export_expiry']))
+		unset($txt['export_profile_data_desc_list']['expiry']);
+	else
+		$txt['export_profile_data_desc_list']['expiry'] = sprintf($txt['export_profile_data_desc_list']['expiry'], $modSettings['export_expiry']);
+
+	$context['export_profile_data_desc'] = sprintf($txt['export_profile_data_desc'], '<li>' . implode('</li><li>', $txt['export_profile_data_desc_list']) . '</li>');
+}
+
+/**
+ * Downloads exported profile data file.
+ */
+function download_export_file($memID)
+{
+	global $modSettings, $maintenance, $context, $txt;
+
+	$export_formats = array(
+		'XML' => array('extension' => 'xml', 'mime' => 'application/xml'),
+		// 'CSV' => array('extension' => 'csv', 'mime' => 'text/csv'),
+		// 'JSON' => array('extension' => 'json', 'mime' => 'application/json'),
+		// 'HTML' => array('extension' => 'html', 'mime' => 'text/html'),
+	);
+
+	// This is done to clear any output that was made before now.
+	ob_end_clean();
+
+	if (!empty($modSettings['enableCompressedOutput']) && !headers_sent() && ob_get_length() == 0)
+	{
+		if (@ini_get('zlib.output_compression') == '1' || @ini_get('output_handler') == 'ob_gzhandler')
+			$modSettings['enableCompressedOutput'] = 0;
+
+		else
+			ob_start('ob_gzhandler');
+	}
+
+	if (empty($modSettings['enableCompressedOutput']))
+	{
+		ob_start();
+		header('content-encoding: none');
+	}
+
+	// No access in strict maintenance mode.
+	if (!empty($maintenance) && $maintenance == 2)
+	{
+		send_http_status(404);
+		exit;
+	}
+
+	// We can't give them anything without these.
+	if (empty($_GET['t']) || empty($_GET['format']) || !isset($export_formats[$_GET['format']]))
```

Comment:
$_GET['format'] not used, I think you mean $_GET['f'] here

Reply:
Er, I don't think so. Unless I missed something, only `$_GET['format']` is used, not `$_GET['f']`.
https://github.com/SimpleMachines/SMF2.1/pull/5081/files#diff-22dc696c41300fca70fca6941cea4df1R3318

Reply:
The logic doesn't work at the moment, it is triggering an HTTP400 .  If you look at it in the debugger, what is passed is not called 'format', it is called 'f'.  

Maybe it's mixed?  But they either all need to go 'format' or 'f'.  

See the screenshot in the comments from ~line 1331.